core/src/cluster_slots.rs

```diff
@@ -192,9 +194,15 @@ impl ClusterSlots {
         let slot_peers = slot_peers.read().unwrap();
         repair_peers
             .iter()
-            .map(|peer| slot_peers.get(&peer.id).cloned().unwrap_or(0))
+            .map(|peer| slot_peers.get(&peer.id).is_some())
             .zip(stakes)
-            .map(|(a, b)| a + b)
+            .map(|(did_complete_slot, peer_stake)| {
+                if did_complete_slot {
+                    max_stake
```

Comment:
I wonder if this should be slightly higher than `max_stake`. The existing behavior for the max staked node with a complete slot would be `2 * max_stake`.

Reply:
hmm, I don't think that's true? We got rid of the `a+b` logic. 

Here's a test that runs this scenario and we end up with only `max_stake`:
https://github.com/solana-labs/solana/blob/2034da17d327e3d74dd3bdf8f2ae6a6ea28c21e8/core/src/cluster_slots.rs#L348-L357

Reply:
The `a+b` logic is being removed so if the highest staked node is in `slot_peers` now it will be normalized to `max_stake` where the existing logic would normalize it to `max_stake + max_stake` right? I'm wondering if we want to do something like:

```
if did_complete_slot {
    peer_stake + max_stake
} else {
    peer_stake
}
```

Which would normalize a node in `slot_peers` to a value at least as large as a the highest staked node, and be more similar to the existing behavior for highly staked nodes in `slot_peers`.

Reply:
ah yeah misunderstood your comment, I thought you meant the current change would cause  the max staked node to have `2 * max_stake`

> Which would normalize a node in slot_peers to a value at least as large as a the highest staked node
Hmm yeah we could do this, although i think the effect will be pretty minimal once we get more than 10 validators that have completed a slot

Reply:
yeah, my thinking is that this would preserve a distinction between a highly staked validator in `slot_peers` and one that is not while also ensuring that a node in `slot_peers` is given a sufficient bump that it won't be lost behind highly staked validators.

Reply:
Added here: https://github.com/solana-labs/solana/pull/28625/commits/fbb752cb2811c58c069e750e973c10643fe55488